app/controllers/challenges/memberships_controller.rb

```diff
@@ -22,5 +18,11 @@ def create
         render :new, status: :unprocessable_entity
       end
     end
+
+    private
+
+    def check_for_membership
+      redirect_back fallback_location: root_path, alert: "You're already a participant of this challenge." if Current.challenge_member?
```

Comment:
Good for now, in the future I can image something like `require_not_challenge_member`